play-json/shared/src/main/scala/play/api/libs/json/JsLookupResult.scala

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2009-2017 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package play.api.libs.json
+
+sealed trait JsLookupResult extends Any with JsReadable {
+  /**
+   * Tries to convert the node into a JsValue
+   */
+  def toOption: Option[JsValue]
+
+  def toEither: Either[JsonValidationError, JsValue]
+
+  def get: JsValue = toOption.get
```

Comment:
In `JsUndefined` this could be something like:
```scala
throw new NoSuchElementException(error)
```

which would be more useful than the generic exception thrown by `Option#get`